lib/index.ts

```diff
@@ -0,0 +1,572 @@
+	_pagination: FrontPagination;
+	_links: FrontLinks;
+	_results: FrontComment[];
+}
+
+export interface FrontConversationInboxes {
+	_pagination: FrontPagination;
+	_links: FrontLinks;
+	_results: FrontInbox[];
+}
+
+export interface FrontConversationFollowers {
+	_pagination: FrontPagination;
+	_links: FrontLinks;
+	_results: FrontAuthor[];
+}
+
+export interface FrontConversationMessages {
+	_pagination: FrontPagination;
+	_links: FrontLinks;
+	_results: FrontMessage[];
+}
+
+export namespace FrontConversationRequest {
+	// Request structures /////////////////////////////////////////////////////
+	export interface List {
+		q?: string;
+		page?: number;
+		limit?: number;
+	}
+
+	export interface Get {
+		conversation_id: string;
+	}
+
+	export interface Update {
+		conversation_id: string;
+		assignee_id?: string;
+		inbox_id?: string;
+		status?: string;
+		tags?: string[];
+	}
+
+	export interface ListComments {
+		conversation_id: string;
+	}
+
+	export interface ListInboxes {
+		conversation_id: string;
+	}
+
+	export interface ListFollowers {
+		conversation_id: string;
+	}
+
+	export interface ListMessages {
+		conversation_id: string;
+		page?: number;
+		limit?: number;
+	}
+}
+
+// Inboxes ////////////////////////////////////////////////////////////////////
+export interface FrontInbox {
+	_links: FrontLinks;
+	address: string;
+	id: string;
+	name: string;
+	send_as: string;
+	type: string;
+}
+
+export interface FrontInboxes {
+	_pagination: FrontPagination;
+	_links: FrontLinks;
+	_results: FrontInbox[];
+}
+
+export interface FrontInboxCreation {
+	id: string;
+	name: string;
+}
+
+export interface FrontInboxChannels {
+	_pagination: FrontPagination;
+	_links: FrontLinks;
+	_results: FrontChannel[];
+}
+
+export interface FrontInboxConversations {
+	_pagination: FrontPagination;
+	_links: FrontLinks;
+	_results: FrontConversation[];
+}
+
+export interface FrontInboxTeammates {
+	_pagination: FrontPagination;
+	_links: FrontLinks;
+	_results: FrontAuthor[];
+}
+
+export namespace FrontInboxRequest {
+	// Request structures /////////////////////////////////////////////////////
+	export interface Create {
+		name: string;
+		teammate_ids?: string[];
+	}
+
+	export interface Get {
+		inbox_id: string;
+	}
+
+	export interface ListChannels {
+		inbox_id: string;
+	}
+
+	export interface ListConversations {
+		inbox_id: string;
+		q?: string;
+		page?: number;
+		limit?: number;
+	}
+
+	export interface ListTeammates {
+		inbox_id: string;
+	}
+}
+
+// Messages ///////////////////////////////////////////////////////////////////
+export interface FrontMessage {
+	_links: FrontLinks;
+	id: string;
+	type: string;
+	is_inbound: boolean;
+	created_at: number;
+	blurb: string;
+	author: FrontAuthor;
+	recipients: FrontRecipient[];
+	body: string;
+	text: string;
+	attachments: FrontAttachment[];
+	metadata: any;
+}
+
+export namespace FrontMessageRequest {
+	// Request structures /////////////////////////////////////////////////////
```

Comment:
Very easy to visually see delineation between the sections (Comments, Messages, etc).

I can remove it if you find it offensive, though. :)